### PR TITLE
Update image ref templating to make registry configurable

### PR DESCRIPTION
**Pull request by ced0ps**
- **This repository allows merge commits, if you're creating a Release, make sure to select 'Squash and merge'**
- **When merging subtree updates, it is important to select 'Create a merge commit' when merging.**

towards https://github.com/giantswarm/giantswarm/issues/27539

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- `Values.image.registry`
+
+### Changed
+
+- Tempalte for image reference in deployment.yaml
```
Comment (Gacko) [inline CHANGELOG.md:16]:
```suggestion
- Template for image reference in deployment.yaml
```

**Review by Gacko — APPROVED** [review]

**Review by paurosello — APPROVED** [review]

**Comment by ubergesundheit** [comment]
Depends on how Phoenix wants to maintain the chart. I think I would change it in the fork, then update from it here. It may be neccessary to set up subtree again as I don't know the exact state of the repository.

**Comment by ced0ps** [comment]
@giantswarm/team-phoenix if you could discuss how you want to handle this repo. If you're happy with the current changes please merge this PR.

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -148,7 +148,7 @@ spec:
         - /controller
         securityContext:
           {{- toYaml .Values.securityContext | nindent 10 }}
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```
Comment (mcharriere) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:151]: I'd suggest to use the schema described [here](https://github.com/giantswarm/schema/blob/main/image/v0.0.1.json)

I've seen that in general we translate `repository` as `registry + name`.

```suggestion
        image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"

```

**Comment by T-Kukawka** [comment]
@ced0ps was this change tested?

File: helm/aws-load-balancer-controller/templates/deployment.yaml (hunk 2 above)
Comment (T-Kukawka) [reply helm/aws-load-balancer-controller/templates/deployment.yaml:151]: in the config repo we have a different scheme: https://github.com/giantswarm/config/blob/5a40fee0b2521707ac773c9b22b3ee78ba27c7a1/installations/giraffe/config.yaml.patch#L50

Comment (mcharriere) [reply helm/aws-load-balancer-controller/templates/deployment.yaml:151]: yeah, but `image.registry` and `global.image.registry` are also there. Not sure where they are coming from.

this is from `argali/pcn01`:
```
$ helm get values -a aws-load-balancer-controller | yq .image
pullPolicy: IfNotPresent
registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
repository: quay.io/giantswarm/aws-alb-ingress-controller
tag: v2.4.1

$ helm get values -a aws-load-balancer-controller | yq .global
image:
  registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
imageRegistry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
```

**Comment by T-Kukawka** [comment]
testing current app version on giraffe:
```
kubectl get cm aws-load-balancer-controller-chart-values -n giantswarm -o yaml
global:
      image:
        registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
      imageRegistry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
    image:
      registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
    provider: aws
    registry:
      domain: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
```

```
kubectl get deployments aws-load-balancer-controller -n aws-load-balancer-controller -o yaml

        image: giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm/aws-alb-ingress-controller:v2.4.1
```

File: helm/aws-load-balancer-controller/values.schema.json [hunk 3]
```diff
@@ -91,6 +91,9 @@
                 "pullPolicy": {
                     "type": "string"
                 },
+                "registry": {
+                    "type": "string"
+                },
                 "repository": {
```
Comment (ced0ps) [inline helm/aws-load-balancer-controller/values.schema.json:97]:
```suggestion
                "name": {
```

**Comment by T-Kukawka** [comment]
testing latest app test-version on giraffe:
```
kubectl get cm aws-load-balancer-controller-chart-values -n giantswarm -o yaml

global:
      image:
        registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
      imageRegistry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
    image:
      registry: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
    provider: aws
    registry:
      domain: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
```

```
kubectl get deployments aws-load-balancer-controller -n aws-load-balancer-controller -o yaml

        image: giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm/aws-alb-ingress-controller:v2.4.1
```

testing latest app test-version on gorilla:
```
kubectl get cm aws-load-balancer-controller-chart-values -n giantswarm -o yaml

no image/registry reference
```

```
kubectl get deployments aws-load-balancer-controller -n aws-load-balancer-controller -o yaml

               image: quay.io/giantswarm/aws-alb-ingress-controller:v2.4.1
```

**Comment by T-Kukawka** [comment]
testing on giraffe 1.3.2 managed (current app)

```
kubectl get deployments aws-load-balancer-controller -n aws-load-balancer-controller -o yaml

               image: quay.io/giantswarm/aws-alb-ingress-controller:v2.4.1
```

**Comment by T-Kukawka** [comment]
cc @ced0ps @paurosello @mcharriere i think we are good to go right?

**Comment by ced0ps** [comment]
@T-Kukawka will you do the release?

**Pull request merged**